ui/lib/core/addon/components/secret-list-header.hbs

```diff
@@ -25,6 +25,80 @@
       <h1 class="title is-3">
         <Icon @name={{@model.icon}} @size="24" class="has-text-grey-light" />
         {{@model.id}}
+
+        {{#if this.isPki}}
+          {{#if @isEngine}}
+            <LinkToExternal @route="secretsListRoot" class="tag is-borderless" data-test-new-pki-beta-button>
+              <Icon @name="arrow-left" />
+              Return to old PKI
+            </LinkToExternal>
+          {{else}}
+            {{#if this.shouldHidePkiBetaModal}}
+              <LinkTo @route="vault.cluster.secrets.backend.pki.overview" class="tag is-success is-borderless">
+                <Icon @name="key" />
+                New PKI UI available
+              </LinkTo>
+            {{else}}
+              <button
+                type="button"
+                class="tag is-success is-v-centered pki-beta-button"
+                {{on "click" (action (mut this.modalOpen) true)}}
+                data-test-old-pki-beta-button
+              >
+                <Icon @name="key" />
+                New PKI UI available
+              </button>
+              <Modal
+                @title="New PKI Beta"
+                @onClose={{action (mut this.modalOpen) false}}
+                @isActive={{this.modalOpen}}
+                @showCloseButton={{true}}
+              >
+                <section class="modal-card-body">
+                  The new PKI beta includes, among other things:
+                  <ul class="pki-beta-modal-list">
+                    <li>Multiple issuers can now be generated and managed in the UI.</li>
+                    <li>Non-disruptive root rotation: quickly rotate a root within the UI without disrupting issuance.</li>
+                    <li>UI upgrades: smoother configuration, an overview page, and more certificate information.</li>
+                  </ul>
+                  You will always be able to return to this version.
+                  <br />
+                  <br />
+                  You can also copy the following URL and bookmark it to always go to the beta version:
+                  <br />
+                  <code class="has-text-danger has-background-white-bis">
+                    http://localhost:8200/ui/vault/secrets/{{@model.id}}/pki/overview
```

Comment:
We probably don't want `localhost` here - I think you can add a getter in `SecretListHeader` for the window origin - maybe with a saying it can be removed along with the beta modal
```js
  get windowOrigin() {
    return window.location.origin;
  }
```

Reply:
Love this suggestion! I was trying to figure out if this was dynamic or not!